precommit/src/main/shell/test-patch.sh

```diff
@@ -2292,40 +2292,121 @@ function check_unittests
+function bugsystem_linecomments_queue
 {
-  declare title=$1
+  declare plugin=$1
   declare fn=$2
   declare line
-  declare bugs
-  declare realline
+  declare linenum
   declare text
+
+  if [[ ! -f "${GITUNIDIFFLINES}" ]]; then
+    return
+  fi
+
+  while read -r line; do
+    file=$(echo "${line}" | cut -f1 -d:)
+    if [[ "${file}" =~ ^\.\/ ]]; then
+      file=${file:2}
+    fi
+    linenum=$(echo "${line}" | cut -f2 -d:)
+    text=$(echo "${line}" | cut -f3- -d:)
+
+    echo "${file}:${linenum}:${plugin}:${text}" >> "${PATCH_DIR}/linecomments-in.txt"
+
+  done < "${fn}"
+}
+
+## @description  Call relevant bug systems to write comments
+## @audience     private
+## @stability    evolving
+## @replaceable  no
+function bugsystem_linecomments_writer
+{
+  declare fn=$1
+  declare linenum=$2
+  shift 2
+  declare -a text
+  text=("$@")
   declare idxline
   declare uniline
 
+  if [[ -z "${fn}" ]]; then
+    return
+  fi
+
+  idxline="${fn}:${linenum}:"
+  uniline=$("${GREP}" "${idxline}" "${GITUNIDIFFLINES}" | cut -f3 -d: )
+
+  for bugs in ${BUGLINECOMMENTS}; do
+    if declare -f "${bugs}_linecomments" >/dev/null;then
+      "${bugs}_linecomments" "${fn}" "${linenum}" "${uniline}" "${linetext[@]}"
+    fi
+  done
+}
+
+## @description  Write all of the bugsystem linecomments
+## @audience     public
+## @stability    evolving
+## @replaceable  no
+function bugsystem_linecomments_trigger
+{
+  declare line
+  declare fn
+  declare linenum
+  declare text
+  declare -a linetext
+  declare prevnum
+  declare prevfn
+  declare plugin
+
   if [[ ! -f "${GITUNIDIFFLINES}" ]]; then
     return
   fi
 
+  # sort the file such that all files and lines are now next to each other
+  sort "${PATCH_DIR}/linecomments-in.txt" > "${PATCH_DIR}/linecomments-sorted.txt"
+
   while read -r line;do
-    file=$(echo "${line}" | cut -f1 -d:)
-    realline=$(echo "${line}" | cut -f2 -d:)
+    fn=$(echo "${line}" | cut -f1 -d:)
+    if [[ -z "${prevfn}" ]]; then
+      prevfn=${fn}
+    fi
+
+    linenum=$(echo "${line}" | cut -f2 -d:)
+    if [[ -z "${prevnum}" ]]; then
+      prevnum=${linenum}
+    fi
+
     text=$(echo "${line}" | cut -f3- -d:)
-    idxline="${file}:${realline}:"
-    uniline=$(${GREP} "${idxline}" "${GITUNIDIFFLINES}" | cut -f3 -d: )
 
-    for bugs in ${BUGLINECOMMENTS}; do
-      if declare -f "${bugs}_linecomments" >/dev/null;then
-        "${bugs}_linecomments" "${title}" "${file}" "${realline}" "${uniline}" "${text}"
-      fi
-    done
-  done < "${fn}"
+    if [[ "${prevfn}" == "${fn}" ]] &&
+       [[ ${prevnum} -eq ${linenum} ]]; then
+      linetext+=("${text}")
+      continue
+    fi
+
+    bugsystem_linecomments_writer "${prevfn}" "${prevnum}" "${linetext[@]}"
+    prevfn=${fn}
+    prevnum=${linenum}
+    linetext=("${text}")
+  done < "${PATCH_DIR}/linecomments-sorted.txt"
+
+  bugsystem_linecomments_writer "${prevfn}" "${prevnum}" "${linetext[@]}"
+}
+
+function trigger_errors
+{
+  echo ${unprotected}
```

Comment:
shellcheck:8: note: Double quote to prevent globbing and word splitting. [SC2086]
shellcheck:8: warning: unprotected is referenced but not assigned. [SC2154]